SpriteBuilder/ccBuilder/ResourceManagerUtil.m

```diff
@@ -305,40 +316,40 @@ + (void)setAttributedTitle:(NSString*)fontName ofMenuItem:(NSMenuItem*)item {
 #pragma mark File icons
 
 + (NSImage*) thumbnailImageForResource:(RMResource*)res {
-    NSImage *image = [res previewForResolution:nil];
+
+    NSString* path = [res absoluteResolutionPath:nil];
     
-    if (![image isValid])
-    {
-        NSLog(@"Invalid Image");
+    CGFloat viewScale = [AppDelegate appDelegate].derivedViewScaleFactor;
+    
+    CGSize size = CGSizeMake(kRMImagePreviewSize*viewScale, kRMImagePreviewSize*viewScale);
+
+    NSURL *fileURL = [NSURL fileURLWithPath:path];
+    
+    if (!path|| !fileURL) {
         return nil;
     }
     
-    [image setScalesWhenResized:YES];
-
-    float maxSideLength = 30.0;
-    float scale = 1;
-    float oldWidth = image.size.width;
-    float oldHeight = image.size.height;
+    CGImageRef ref = QLThumbnailImageCreate(kCFAllocatorDefault,
+                                            (__bridge CFURLRef)fileURL,
+                                            CGSizeMake(size.width, size.height),
+                                            nil);
     
-    if (oldWidth > maxSideLength || oldHeight > maxSideLength) {
-        if (oldWidth > oldHeight) {
-            scale = (oldWidth / maxSideLength);
-        } else scale = (oldHeight / maxSideLength);
+    if (ref != NULL) {
+        NSBitmapImageRep *bitmapImageRep = [[NSBitmapImageRep alloc] initWithCGImage:ref];
+        NSImage *newImage = nil;
+        
+        if (bitmapImageRep) {
+            newImage = [[NSImage alloc] initWithSize:[bitmapImageRep size]];
+            [newImage addRepresentation:bitmapImageRep];
+            CFRelease(ref);
```

Comment:
The ref is set but the release relies on the bitmapImageRep to be set, potential leak.